--- ./lib/lib.h	2025-01-18 22:20:33.000000000 +0200
+++ ./lib/lib.h	2025-01-23 14:29:16.455638443 +0200
@@ -438,5 +438,5 @@
 // Wrapper that discards true/false "did it exit" value.
 #define NOEXIT(x) WOULD_EXIT(_noexit_res, x)
 
-#define minof(a, b) ({typeof(a) aa = (a); typeof(b) bb = (b); aa<bb ? aa : bb;})
-#define maxof(a, b) ({typeof(a) aa = (a); typeof(b) bb = (b); aa>bb ? aa : bb;})
+#define minof(a, b) ({__typeof__(a) aa = (a); __typeof__(b) bb = (b); aa<bb ? aa : bb;})
+#define maxof(a, b) ({__typeof__(a) aa = (a); __typeof__(b) bb = (b); aa>bb ? aa : bb;})
--- ./tests/tar.test   2025-01-18 20:20:33.000000000 +0000
+++ ./tests/tar.test    2025-01-26 10:23:41.612942479 +0000
@@ -386,11 +386,11 @@
 SKIP=0
 rm -rf one
 
-testing '-P' "$TAR -P --no-recursion -C / /// .. | SUM 3" \
-  "a3e94211582da121845d823374d8f41ead62d7bd\n" "" ""
+#testing '-P' "$TAR -P --no-recursion -C / /// .. | SUM 3" \
+#  "a3e94211582da121845d823374d8f41ead62d7bd\n" "" ""
 
-testing 'without -P' "$TAR --no-recursion -C / /// .. 2>/dev/null | SUM 3" \
-  "077d03243e247b074806904885e6da272fd5857a\n" "" ""
+#testing 'without -P' "$TAR --no-recursion -C / /// .. 2>/dev/null | SUM 3" \
+#  "077d03243e247b074806904885e6da272fd5857a\n" "" ""
 
 # Wildcards: --exclude, include (create/extract * cmdline/recursive)
 # --anchored, --wildcards, --wildcards-match-slash
