--- ./test/function2.c	2024-12-06 02:43:47.000000000 +0200
+++ ./test/function2.c	2025-01-24 14:46:43.543004597 +0200
@@ -189,7 +189,8 @@
 
 Aligned1024 s = {.c = 77};
 
-  ASSERT(257, struct_test131(
+// Kefir does not support -Wno-psabi
+/*  ASSERT(257, struct_test131(
     g[0], g[1], g[2], g[3], g[4],
     f[0], f[1], f[2], f[3], f[4], f[5],
     (int) 11,
@@ -202,7 +203,7 @@
     (double) 6.7,
     (double) 7.8,
     s, // (Aligned1024)
-    (long double) 11.1));
+    (long double) 11.1));*/
 
   ASSERT(257, struct_test130(
     g[0], g[1], g[2], g[3], g[4],
