diff -ru gcc/Makefile.in gcc/Makefile.in
--- gcc/Makefile.in	2013-07-08 12:15:35.000000000 +0300
+++ gcc/Makefile.in	2025-10-06 21:53:28.562709034 +0300
@@ -1724,7 +1724,7 @@
 quickstrap: all
 	cd $(toplevel_builddir) && $(MAKE) all-target-libgcc
 
-all.internal: start.encap rest.encap doc
+all.internal: start.encap rest.encap
 # This is what to compile if making a cross-compiler.
 all.cross: native gcc-cross$(exeext) cpp$(exeext) specs \
 	libgcc-support lang.all.cross doc @GENINSRC@ srcextra
diff -ru libgcc/config/i386/linux-unwind.h libgcc/config/i386/linux-unwind.h
--- libgcc/config/i386/linux-unwind.h	2012-04-20 10:51:30.000000000 +0300
+++ libgcc/config/i386/linux-unwind.h	2025-10-06 21:54:22.589002843 +0300
@@ -53,7 +53,7 @@
   if (*(unsigned char *)(pc+0) == 0x48
       && *(unsigned long long *)(pc+1) == RT_SIGRETURN_SYSCALL)
     {
-      struct ucontext *uc_ = context->cfa;
+      ucontext_t *uc_ = context->cfa;
       /* The void * cast is necessary to avoid an aliasing warning.
          The aliasing warning is correct, but should not be a problem
          because it does not alias anything.  */
@@ -142,7 +142,7 @@
 	siginfo_t *pinfo;
 	void *puc;
 	siginfo_t info;
-	struct ucontext uc;
+	ucontext_t uc;
       } *rt_ = context->cfa;
       /* The void * cast is necessary to avoid an aliasing warning.
          The aliasing warning is correct, but should not be a problem
