diff -ru tests/tail/inotify-dir-recreate.sh ./tests/tail/inotify-dir-recreate.sh
--- tests/tail/inotify-dir-recreate.sh	2026-02-04 13:45:38.000000000 +0200
+++ tests/tail/inotify-dir-recreate.sh	2026-03-12 20:57:40.980059364 +0200
@@ -22,6 +22,8 @@
 print_ver_ tail
 require_inotify_supported_
 
+skip_ "Fails in overlayfs container"
+
 # Terminate any background tail process
 cleanup_() { kill $pid 2>/dev/null && wait $pid; }
 
diff -ru tests/tail/inotify-rotate-resources.sh ./tests/tail/inotify-rotate-resources.sh
--- tests/tail/inotify-rotate-resources.sh	2026-02-04 13:45:38.000000000 +0200
+++ tests/tail/inotify-rotate-resources.sh	2026-03-12 20:57:57.160059440 +0200
@@ -27,6 +27,8 @@
 
 require_strace_ 'inotify_add_watch,inotify_rm_watch'
 
+skip_ 'Fails in overlayfs container'
+
 check_tail_output()
 {
   local delay="$1"
