--- ./test/test-udp-multicast-join6.c	2024-10-18 22:02:38.000000000 +0300
+++ ./test/test-udp-multicast-join6.c	2025-01-10 19:24:10.118209496 +0200
@@ -169,6 +169,8 @@
   int r;
   struct sockaddr_in6 addr;
 
+  RETURN_SKIP("NTest does not work with Kefir");
+
   if (!can_ipv6_external())
     RETURN_SKIP("No external IPv6 interface available");
 
--- ./test/test-udp-multicast-join.c	2024-10-18 22:02:38.000000000 +0300
+++ ./test/test-udp-multicast-join.c	2025-01-10 19:24:31.498300797 +0200
@@ -138,6 +138,7 @@
 
 
 TEST_IMPL(udp_multicast_join) {
+  RETURN_SKIP("Test does not work with Kefir");
 #if defined(__OpenBSD__)
   RETURN_SKIP("Test does not currently work in OpenBSD");
 #endif
