--- literals.lua	2023-05-02 22:02:31.000000000 +0200
+++ literals.lua	2023-07-16 11:23:46.737407733 +0200
@@ -294,7 +294,7 @@
 
 
 -- testing decimal point locale
-if os.setlocale("pt_BR") or os.setlocale("ptb") then
+if nil and (os.setlocale("pt_BR") or os.setlocale("ptb")) then
   assert(tonumber("3,4") == 3.4 and tonumber"3.4" == 3.4)
   assert(tonumber("  -.4  ") == -0.4)
   assert(tonumber("  +0x.41  ") == 0X0.41)
