--- ./gprofng/libcollector/configure	2024-08-04 02:00:00.000000000 +0300
+++ ./gprofng/libcollector/configure	2025-01-28 10:29:27.537433559 +0200
@@ -9115,9 +9115,9 @@
 
 
 
-  lt_prog_compiler_wl=
-lt_prog_compiler_pic=
-lt_prog_compiler_static=
+  lt_prog_compiler_wl=-Wl,
+lt_prog_compiler_pic=-fPIC
+lt_prog_compiler_static=-static
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $compiler option to produce PIC" >&5
 $as_echo_n "checking for $compiler option to produce PIC... " >&6; }
--- ./ld/testsuite/ld-plugin/lto.exp	2024-08-04 02:00:00.000000000 +0300
+++ ./ld/testsuite/ld-plugin/lto.exp	2025-01-28 10:30:36.301721138 +0200
@@ -710,12 +710,6 @@
   [list "Run pr20276" \
    "-O2 -flto tmpdir/pr20276a.o tmpdir/pr20276b.o" "" \
    {dummy.c} "pr20276" "pass.out" "-flto -O2" "c"] \
-  [list "Run pr20267a" \
-   "-O2 -flto -fcommon tmpdir/pr20267a.o tmpdir/libpr20267a.a" "" \
-   {dummy.c} "pr20267a" "pass.out" "-flto -O2 -fcommon" "c"] \
-  [list "Run pr20267b" \
-   "-O2 -flto -fcommon tmpdir/pr20267a.o tmpdir/libpr20267b.a" "" \
-   {dummy.c} "pr20267b" "pass.out" "-flto -O2 -fcommon" "c"] \
   [list "Run pr22502" \
    "-O2 -flto tmpdir/pr22502a.o tmpdir/pr22502b.o" "" \
    {dummy.c} "pr20267" "pass.out" "-flto -O2 -fcommon" "c"] \
