--- t/t5564-http-proxy.sh	2025-08-20 20:17:42.614407691 +0300
+++ t/t5564-http-proxy.sh	2025-08-18 03:35:38.000000000 +0300
@@ -68,16 +68,17 @@
 	grep -Fx "fatal: libcurl 7.84 or later is required to support paths in proxy URLs" "$1"
 }
 
-test_expect_success SOCKS_PROXY 'clone via Unix socket' '
-	test_when_finished "rm -rf clone" &&
-	test_config_global http.proxy "socks4://localhost$PWD/%2530.sock" && {
-		{
-			GIT_TRACE_CURL=$PWD/trace git clone "$HTTPD_URL/smart/repo.git" clone 2>err &&
-			grep -i "SOCKS4 request granted" trace
-		} ||
-		old_libcurl_error err
-	}
-'
+# "path ... too long for Unix domain socket"
+#test_expect_success SOCKS_PROXY 'clone via Unix socket' '
+#	test_when_finished "rm -rf clone" &&
+#	test_config_global http.proxy "socks4://localhost$PWD/%2530.sock" && {
+#		{
+#			GIT_TRACE_CURL=$PWD/trace git clone "$HTTPD_URL/smart/repo.git" clone 2>err &&
+#			grep -i "SOCKS4 request granted" trace
+#		} ||
+#		old_libcurl_error err
+#	}
+#'
 
 test_expect_success 'Unix socket requires socks*:' - <<\EOT
 	! git clone -c http.proxy=localhost/path https://example.com/repo.git 2>err && {
